experimental/segmenter/src/rule_segmenter.rs

```diff
@@ -26,10 +26,10 @@ pub enum RuleStatusType {
 /// encoding methods and granularity such as grapheme cluster, word, etc.
 pub trait RuleBreakType<'l, 's> {
```

Comment:
oh, while we're at it, we should migrate these lifetimes onto the methods, it should be possible to bound marker types on a simple trait without the lifetimes leaking

Reply:
(no need to do this if the traits are staying private for now)